pkg/certificates/renewerbottlerocket.go

```diff
@@ -0,0 +1,358 @@
+
+	return nil
+}
+
+func (b *BottlerocketRenewer) CopyEtcdCerts(ctx context.Context, node string, ssh SSHRunner) error {
+	logger.V(4).Info("Reading certificate from ETCD node", "node", node)
+	logger.V(4).Info("Using backup directory", "path", b.backup)
+
+	remoteTempDir := brTempDir
+
+	if _, err := ssh.RunCommand(ctx, node, buildBRSheltieCmd(
+		buildBREtcdCopyCertsToTmpCmd(remoteTempDir),
+	)); err != nil {
+		return fmt.Errorf("copying certificates to tmp: %v", err)
+	}
+
+	if _, err := ssh.RunCommand(ctx, node, buildBRListTmpFilesCmd(remoteTempDir)); err != nil {
+		return fmt.Errorf("listing certificate files: %v", err)
+	}
+
+	crtContent, err := ssh.RunCommand(ctx, node, buildBRReadTmpCertCmd(remoteTempDir))
+	if err != nil {
+		return fmt.Errorf("reading certificate file: %v", err)
+	}
+
+	if len(crtContent) == 0 {
+		return fmt.Errorf("certificate file is empty")
+	}
+
+	logger.V(4).Info("Reading key from ETCD node", "node", node)
+
+	keyContent, err := ssh.RunCommand(ctx, node, buildBRReadTmpKeyCmd(remoteTempDir))
+	if err != nil {
+		return fmt.Errorf("read key file: %v", err)
+	}
+	if len(keyContent) == 0 {
+		return fmt.Errorf("key file is empty")
+	}
+
+	destDir := filepath.Join(b.backup, tempLocalEtcdCertsDir)
+	if err := os.MkdirAll(destDir, 0o700); err != nil {
+		return fmt.Errorf("create local cert dir: %v", err)
+	}
+
+	crtPath := filepath.Join(destDir, "apiserver-etcd-client.crt")
+	keyPath := filepath.Join(destDir, "apiserver-etcd-client.key")
+
+	logger.V(4).Info("Writing certificates to:")
+	logger.V(4).Info("Certificate", "path", crtPath)
+	logger.V(4).Info("Key", "path", keyPath)
+
+	if err := os.WriteFile(crtPath, []byte(crtContent), 0o600); err != nil {
+		return fmt.Errorf("write certificate file: %v", err)
+	}
+	if err := os.WriteFile(keyPath, []byte(keyContent), 0o600); err != nil {
+		return fmt.Errorf("write key file: %v", err)
+	}
+
+	if _, err := ssh.RunCommand(ctx, node, buildBRSheltieCmd(
+		buildBREtcdCleanupTmpCmd(remoteTempDir),
+	)); err != nil {
+		return fmt.Errorf("cleanup temporary files: %v", err)
+	}
+
+	logger.V(4).Info("Certificates copied successfully")
+	logger.V(4).Info("Backup directory", "path", b.backup)
+	logger.V(4).Info("Certificate path", "path", crtPath)
+	logger.V(4).Info("Key path", "path", keyPath)
+
+	return nil
+}
+
+func buildBRSheltieCmd(commands ...string) string {
+	script := strings.Join(commands, "\n")
+
+	fullCommand := fmt.Sprintf("sudo sheltie << 'EOF'\nset -euo pipefail\n%s\nEOF", script)
+	return fullCommand
+}
+
+func buildBRImagePullCmd() string {
+	return `IMAGE_ID=$(apiclient get | apiclient exec admin jq -r '.settings["host-containers"]["kubeadm-bootstrap"].source')
+ctr image pull ${IMAGE_ID}`
+}
+
+func buildBRControlPlaneBackupCertsCmd(_ string, hasExternalEtcd bool, backupDir, certDir string) string {
+	var script string
+	if hasExternalEtcd {
+		script = fmt.Sprintf(`mkdir -p '/etc/kubernetes/pki.bak_%[1]s'
+cp -r %[2]s/* '/etc/kubernetes/pki.bak_%[1]s/'
```

Comment:
can we copy it to the `/var/lib/kubeadm/pki.bak%[1]s/` since for bottlerocket keys and certificates are stored under `/var/lib/kubeadm/pki`?